File: Heat/ViewModels/WorkActions/CreateWorkActionViewModel.vb
Imports System.ComponentModel.DataAnnotations

Namespace ViewModels.WorkActions
    Public Class CreateWorkActionViewModel

        <Display(name:="Impianto")> _
        Property PlantID As Integer
        <Display(name:="Riferita all'impianto")> _
        Property PlantDescription As String
        Property PlantList As IEnumerable(Of SelectListItem)
        Property PlantIDSelected As Boolean

        <Display(name:="Data di esecuzione")> _
        <DataType(DataType.Date)> _
        <DisplayFormat(DataFormatString:="{0:yyyy-MM-dd}", ApplyFormatInEditMode:=True)> _
        Property ActionDate As DateTime

        <Display(name:="Tipo di operazione")> _
        Property OperationID As Integer
        Property OperationList As IEnumerable(Of SelectListItem)

        <Display(name:="Operatore assegnato")> _
        Property AssignedOperatorID As Integer
        Property AssignedOperatorList As IEnumerable(Of SelectListItem)

        <Display(name:="Tipo di operazione")> _
        Property TypeID As Integer
        Property TypeList As IEnumerable(Of SelectListItem)

    End Class
End Namespace

File: Heat.ConvertedToC#/ViewModels/WorkActions/CreateWorkActionViewModel.cs
using System;
using System.Collections.Generic;
using System.Web.Mvc;
using System.ComponentModel.DataAnnotations;

namespace Heat.ViewModels.WorkActions
{
    public class CreateWorkActionViewModel
	{

		[Display(Name = "Impianto")]
		public int PlantID { get; set; }
		[Display(Name = "Riferita all'impianto")]
		public string PlantDescription { get; set; }
		public IEnumerable<SelectListItem> PlantList { get; set; }
		public bool PlantIDSelected { get; set; }

		[Display(Name = "Data di esecuzione")]
		[DataType(DataType.Date)]
		[DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
		public DateTime ActionDate { get; set; }

		[Display(Name = "Tipo di operazione")]
		public int OperationID { get; set; }
		public IEnumerable<SelectListItem> OperationList { get; set; }

		[Display(Name = "Operatore assegnato")]
		public int AssignedOperatorID { get; set; }
		public IEnumerable<SelectListItem> AssignedOperatorList { get; set; }

		[Display(Name = "Tipo di operazione")]
		public int TypeID { get; set; }
		public IEnumerable<SelectListItem> TypeList { get; set; }

	}
}

